tests/Middleware/RequestValidationTest.php

```diff
@@ -27,14 +32,70 @@ public function handleRegistersResultInstanceInContainer(): void
                 'body' => '',
             ]
         );
+        $apiProblemBuilder = self::createStub(ApiProblemBuilder::class);
         $container = self::createMock(Container::class);
-        $sut = new RequestValidation(__DIR__ . '/../fixtures/petstore-expanded.json', $container);
+        $sut = new RequestValidation(__DIR__ . '/../fixtures/petstore-expanded.json', $apiProblemBuilder, $container);
 
         $container->expects(self::once())
             ->method('instance')
             ->with(Result::class, $expected);
 
-        $sut->handle(Request::create('/pets?limit=5&tags[]=cat&tags[]=tabby'), fn($var) => new Response());
+        $sut->handle(Request::create($url), fn($var) => new Response());
+    }
+
+    public function dataSetsThatThrowCannotProcessRequest(): array
+    {
+        return [
+            'path not found, empty apiResponseTypes' => [
+                '/hats',
+                Method::GET,
+                [],
+                '{"title":"Not Found","type":"about:blank","status":404,"detail":"\/hats does not match any specified paths in petstore-expanded.json"}',
+                404,
+            ],
+            'path not found, no applicable apiResponseType' => [
+                '/hats',
+                Method::GET,
+                [200 => 'OK'],
+                '{"title":"Not Found","type":"about:blank","status":404,"detail":"\/hats does not match any specified paths in petstore-expanded.json"}',
+                404,
+            ],
+            'path not found, applicable apiResponseType' => [
+                '/hats',
+                Method::GET,
+                [404 => 'Not Found'],
+                '{"title":"Not Found","type":"Not Found","status":404,"detail":"\/hats does not match any specified paths in petstore-expanded.json"}',
+                404,
+            ],
+            'operation not found, applicable apiResponseType' => [
+                '/pets',
+                Method::DELETE,
+                [404 => 'Not Found', 405 => 'Method Not Allowed'],
+                '{"title":"Method Not Allowed","type":"Method Not Allowed","status":405,"detail":"delete operation not specified on path"}',
+                405,
+            ],
+            // TODO test 406 from unsupported content-types once Membrane is reading content-types from requests
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider dataSetsThatThrowCannotProcessRequest
+     */
+    public function catchesCannotProcessRequest(): void
```

Comment:
you have specified a data provider here, but don't seem to be taking in the different vars from the provider or using them in the test?

Reply:
Resolved in tandem with previous comment, apologies that was a careless mistake on my part.